encoding/encodeInto.any.js

```diff
@@ -73,65 +73,74 @@
-        if (destinationFiller === "random") {
-          byte = Math.floor(Math.random() * 256);
+  ].forEach((destinationData) => {
+    ["ArrayBuffer", "SharedArrayBuffer"].forEach((arrayBufferOrSharedArrayBuffer) => {
+      test(() => {
+        // Setup
+        const bufferLength = testData.destinationLength + destinationData.bufferIncrease,
+              destinationOffset = destinationData.destinationOffset,
+              destinationLength = testData.destinationLength,
+              destinationFiller = destinationData.filler,
+              encoder = new TextEncoder(),
+              buffer = new self[arrayBufferOrSharedArrayBuffer](bufferLength),
+              view = new Uint8Array(buffer, destinationOffset, destinationLength),
+              fullView = new Uint8Array(buffer),
+              control = new Array(bufferLength);
+        let byte = destinationFiller;
+        for (let i = 0; i < bufferLength; i++) {
+          if (destinationFiller === "random") {
+            byte = Math.floor(Math.random() * 256);
+          }
+          control[i] = byte;
+          fullView[i] = byte;
         }
-        control[i] = byte;
-        fullView[i] = byte;
-      }
 
-      // It's happening
-      const result = encoder.encodeInto(testData.input, view);
+        // It's happening
+        const result = encoder.encodeInto(testData.input, view);
 
-      // Basics
-      assert_equals(view.byteLength, destinationLength);
-      assert_equals(view.length, destinationLength);
+        // Basics
+        assert_equals(view.byteLength, destinationLength);
+        assert_equals(view.length, destinationLength);
 
-      // Remainder
-      assert_equals(result.read, testData.read);
-      assert_equals(result.written, testData.written.length);
-      for (let i = 0; i < bufferLength; i++) {
-        if (i < destinationOffset || i >= (destinationOffset + testData.written.length)) {
-          assert_equals(fullView[i], control[i]);
-        } else {
-          assert_equals(fullView[i], testData.written[i - destinationOffset]);
+        // Remainder
+        assert_equals(result.read, testData.read);
+        assert_equals(result.written, testData.written.length);
+        for (let i = 0; i < bufferLength; i++) {
+          if (i < destinationOffset || i >= (destinationOffset + testData.written.length)) {
+            assert_equals(fullView[i], control[i]);
+          } else {
+            assert_equals(fullView[i], testData.written[i - destinationOffset]);
+          }
         }
-      }
-    }, "encodeInto() with " + testData.input + " and destination length " + testData.destinationLength + ", offset " + destinationData.destinationOffset + ", filler " + destinationData.filler);
+      }, "encodeInto() into "  + self[arrayBufferOrSharedArrayBuffer].name + " with " + testData.input + " and destination length " + testData.destinationLength + ", offset " + destinationData.destinationOffset + ", filler " + destinationData.filler);
```

Comment:
You can use `arrayBufferOrSharedArrayBuffer` on its own here now (and elsewhere).

Reply:
Done!